tests/foreman/sys/test_pulp3_filesystem.py

```diff
@@ -0,0 +1,92 @@
+"""Tests for Pulp3 file system.
+
+:Requirement: Pulp3 installed
+
+:CaseAutomation: Automated
+
+:CaseLevel: System
+
+:CaseComponent: Pulp
+
+:Assignee: ltran
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+
+"""
+import pytest
+
+from robottelo.ssh import get_connection
+
+
+@pytest.fixture
```

Comment:
> =========================================== 7 passed in 32.25s ==================================

Sadly it does come with a price. This very likely comes from `get_connection` which is now called more often. You can consider making the `connection` fixture a session fixture if it is reusable. I think [pytest-xdist](https://pypi.org/project/pytest-xdist) uses multiple processes so thread safety isn't a huge concern. If [pytest-parallel](https://pypi.org/project/pytest-parallel) is used, it may be. The documentation says it uses a separate channel for every command: https://docs.paramiko.org/en/stable/api/client.html#paramiko.client.SSHClient.exec_command. That means it's probably safe.

I'd try this to see what kind of performance impact it has:

```suggestion
@pytest.fixture(scope="session")
```

Reply:
OK, it is faster: = 7 passed in 21.22s =